pkg/kubelet/kubelet_node_status.go

```diff
@@ -579,7 +579,12 @@ func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error
 	}
 
 	node, changed := kl.updateNode(ctx, originalNode)
-	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency
```

Comment:
LGTM, just one question, maybe for @SergeyKanzhelev or other kubelet code maintainers, If we just want to add a constant to the period `kl.nodeStatusReportFrequency`, why don't we just add it here with a sync.Once so we avoid having to plumb the variable through different layers? 
This will make this solution specific of this control loop, and avoid to expose to the rest of the codebase

Reply:
kubelet starts 2 goroutines in https://github.com/kubernetes/kubernetes/blob/b240c51a725d6c43221709c9b7a04f937757af37/pkg/kubelet/kubelet.go#L1677-L1684.
Both can execute the function `tryUpdateNodeStatus `. `sync.Once` won't give what we want if there are 2 node status update in a short period of time.

Reply:
I see, I understand now , we reset in "each" update , not only once, thanks